controllers/account/account_controller.go

```diff
@@ -36,7 +34,8 @@ import (
 )
 
 var log = logf.Log.WithName("controller_account")
-var AssumeRole = AssumeRoleFunc
+var AssumeRole = awsclient.AssumeRoleFunc
+var assumeRole = awsclient.AssumeRole
```

Comment:
This is confusing - what is the difference between these two things? Can these be named better?

Reply:
In the original code one function was "AssumeRoleFunc" and the other just "assumeRole".  "AssumeRoleFunc"  is basically a wrapper function for `assumeRole`. `assumeRole` handles the logic for assuming an AWS role and obtaining temporary credentials. I can either do 1 or 2

1. update line 37 and add "Func" to the end of "AssumeRole" to keep it somewhat consistent with the original code and add comments like you suggested to describing what each ones does.

2. change "AssumeRoleFunc" to "AssumeRoleAndCreateClient" and "assumeRole" to "handleRoleAssumption" or any other names that you think better encapsulate there purpose.

ref: https://github.com/reedcort/aws-account-operator/blob/OSD-17714/pkg/awsclient/sts_assume_role.go#L78-L153

Reply:
Let's go with Option two here and rename them :)